ocrd_models/ocrd_models/ocrd_mets.py

```diff
@@ -311,20 +445,41 @@ def add_file(self, fileGrp, mimetype=None, url=None, ID=None, pageId=None, force
         if not REGEX_FILE_ID.fullmatch(fileGrp):
             raise ValueError("Invalid syntax for mets:fileGrp/@USE %s (not an xs:ID)" % fileGrp)
         log = getLogger('ocrd_models.ocrd_mets.add_file')
+
+        """
+        # Note: we do not benefit enough from having 
+        # a separate cache for fileGrp elements
+
+        if self._cache_flag: 
+            if fileGrp in self._fileGrp_cache:
+                el_fileGrp = self._fileGrp_cache[fileGrp]
+        """
+
```

Comment:
I don't think caching would help in this place but we could benefit minimally in `add_file_group`, if we determine that the `fileGrp` to add is already a key in `self._file_cache`, we could skip the element exist/create code there.

```suggestion
```